cli/command/swarm/init_test.go

```diff
@@ -123,3 +126,31 @@ func TestSwarmInit(t *testing.T) {
 		golden.Assert(t, cli.OutBuffer().String(), fmt.Sprintf("init-%s.golden", tc.name))
 	}
 }
+
+func TestGetDefaultAddrPool(t *testing.T) {
+
+	var addrList []*net.IPNet
+	// Test only CIRD input and make sure default subnet size is returned
+	addrList, size, err := getDefaultAddrPool("20.20.0.0/16,30.30.0.0/16")
+	assert.NilError(t, err)
+	assert.Check(t, is.Equal(size, 24))
+	assert.Check(t, is.Equal(len(addrList), 2))
+
+	//Pass wrong syntax params
+	_, _, err = getDefaultAddrPool("20.20.0.0/16;30.30.0.0/16;24")
+	assert.Check(t, err != nil)
+
+	// Pass subnet size smaller than subnet
+	_, _, err = getDefaultAddrPool("20.20.0.0/24,30.30.0.0/24:20")
+	assert.Check(t, err != nil)
+
+	// Pass CIDR and subnet with white space and make sure everything works
+	addrList, _, err = getDefaultAddrPool(" 20.20.0.0/20 , 30.30.0.0/20 , 40.40.0.0/16 : 24")
+	assert.NilError(t, err)
+	assert.Check(t, is.Equal(len(addrList), 3))
+
+	// Pass CIDR just with IP address and it should fail
+	_, _, err = getDefaultAddrPool(" 20.20.0.0, 30.30.0.0/20 : 24")
+	assert.Check(t, err != nil)
+
```

Comment:
maybe add a check where the list should succeed but verify that the size parsed correctly and was not default?

Reply:
Should there be a test where the subnet size == the pool allocation size?